qiskit/quantum_info/operators/channel/kraus.py

```diff
@@ -406,7 +406,7 @@ def _tensor_product(self, other, reverse=False):
         if reverse:
             input_dims = self.input_dims() + other.input_dims()
             output_dims = self.output_dims() + other.output_dims()
-            kab_l = [np.kron(b, a) for a in ka_l for b in kb_l]
+            kab_l = [np.kron(a, b) for b in ka_l for a in kb_l]
```

Comment:
This seems more confusing now (`for b in ka_l for a in kb_l`). `arguments-out-of-order` is failing because `np.kron` is defined as `def kron(a,b)`, but the argument names aren't really meaningful in this case. Maybe @chriseclectic has thoughts.

Reply:
Yeah, it was either this or disable the rule. I agree it's kind of confusing I'm good with either approach but figured it was easier to just match the variable names to the parameter names.

A third option would be to use something besides `a` and `b` here so that pylint doesn't think we're passing in by name.

Reply:
Ok, updated in: https://github.com/Qiskit/qiskit-terra/pull/3635/commits/689c413d276bb5c1934b4537964d8cd331a0bd54 to use the original ordering but changed the variable names to be `a_in` and `b_in` so pylint doesn't complain